lld/test/ELF/why-live.s

```diff
@@ -0,0 +1,153 @@
+# REQUIRES: x86
+
+# RUN: llvm-mc -n -filetype=obj -triple=x86_64 %s -o %t.o
+# RUN: echo -e ".globl test_shared\n .section .test_shared,\"ax\",@progbits\n test_shared: jmp test_shared" |\
+# RUN:   llvm-mc -n -filetype=obj -triple=x86_64 -o %t.shared.o
+# RUN: ld.lld -shared %t.shared.o -o %t.so
+
+## Simple live section
+.globl _start
+.section ._start,"ax",@progbits
+_start:
+  jmp test_simple
+  .quad .Lanonymous
+  .quad .Lanonymous_within_symbol
+  jmp test_shared
+  .quad test_local
+.size _start, .-_start
+
+.globl test_simple
+.section .test_simple,"ax",@progbits
+test_simple:
+  jmp test_simple
+  jmp test_from_unsized
+
+# RUN: ld.lld %t.o %t.so -o /dev/null --gc-sections --why-live=test_simple | FileCheck %s --check-prefix=SIMPLE
+
+# SIMPLE:      live symbol: {{.*}}.o:(test_simple)
+# SIMPLE-NEXT: >>> referenced by: {{.*}}.o:(_start) (entry point)
+# SIMPLE-NOT:  >>>
+
+## Live only by being a member of .test_simple
+.globl test_incidental
+test_incidental:
+  jmp test_incidental
+
+# RUN: ld.lld %t.o %t.so -o /dev/null --gc-sections --why-live=test_incidental | FileCheck %s --check-prefix=INCIDENTAL
+
+# INCIDENTAL:      live symbol: {{.*}}.o:(test_incidental)
+# INCIDENTAL-NEXT: >>> in live section: {{.*}}.o:(.test_simple)
+# INCIDENTAL-NEXT: >>> contained live symbol: {{.*}}.o:(test_simple)
+# INCIDENTAL-NEXT: >>> referenced by: {{.*}}.o:(_start) (entry point)
+# INCIDENTAL-NOT:  >>>
+
+## Reached from a reference in section .test_simple directly, since test_simple is an unsized symbol.
+.globl test_from_unsized
+.section .test_from_unsized,"ax",@progbits
+test_from_unsized:
+  jmp test_from_unsized
+
+# RUN: ld.lld %t.o %t.so -o /dev/null --gc-sections --why-live=test_from_unsized | FileCheck %s --check-prefix=FROM-UNSIZED
+
+# FROM-UNSIZED:      live symbol: {{.*}}.o:(test_from_unsized)
+# FROM-UNSIZED-NEXT: >>> referenced by: {{.*}}.o:(.test_simple)
+# FROM-UNSIZED-NEXT: >>> contained live symbol: {{.*}}.o:(test_simple)
+# FROM-UNSIZED-NEXT: >>> referenced by: {{.*}}.o:(_start) (entry point)
+# FROM-UNSIZED-NOT:  >>>
```

Comment:
could replace these `-NOT: >>>` with `-EMPTY:`

Reply:
Done.